trust/meson.build

```diff
@@ -142,12 +142,25 @@ if get_option('test')
     'test-jks'
   ]
 
+  # Some tests fail to link on macOS because they need p11_library_mutex, but
+  # it isn't included unless libp11_kit_testable is linked with
+  # -Wl,--whole-archive or -Wl,-force_load.
+  trust_tests_whole = [
+    'test-index',
+    'test-builder',
+    'test-token'
+  ]
+
   foreach name : trust_tests
+    libp11 = libp11_library_dep
+    if trust_tests_whole.contains(name)
+      libp11 = libp11_library_dep.as_link_whole()
+    endif
```

Comment:
nit: can this be simply written with a [ternary operator](https://mesonbuild.com/Syntax.html#ternary-operator)?